### ISSUE TITLE
Make GPT2ForSequenceClassification computationllay efficient

### ISSUE THREAD

**Issue by Kyeongpil** [huggingface/transformers#22113]
### Feature request

Compute the logits using only the hidden state for the last position of the input sequence. 

### Motivation

The current GPT2ForSequenceClassification module computes logits using all hidden states, which computationally cost is proportional to the length of the input sequence.
https://github.com/huggingface/transformers/blob/v4.26.1/src/transformers/models/gpt2/modeling_gpt2.py#L1384
If we compute the logits using only the hidden state for the last position of the input sequence, the cost is not proportional to the length.

It can not be only applied to , but also to other XXXForSequenceClassification.

### Your contribution

The followings are the code changes that I suggest:
https://github.com/Kyeongpil/transformers/commit/f97f6e38f444522a55f236b37ca70b4e35096e12
